app/Models/Foundation/Summit/Speakers/PresentationSpeaker.php

```
@@ -2309,6 +2309,30 @@ public function hasAnnouncementEmailTypeSent(Summit $summit, string $type): bool
+     * @return bool
+     */
+    public function hasAnnouncementEmailTypeSentSince(Summit $summit, string $type, \DateTime $since): bool
+    {
+        $criteria = Criteria::create();
+
+        $criteria
+            ->where(Criteria::expr()->eq('summit', $summit))
+            ->andWhere(Criteria::expr()->eq('type', $type))
+            ->andWhere(Criteria::expr()->gte('send_date', $since));
+
+        return $this->announcement_summit_emails->matching($criteria)->count() > 0;
```

PHPStan (2.2.8)

[error] Cannot call method matching() on array<models\summit\SpeakerAnnouncementSummitEmail>.


(method.nonObject)